cpu/stm32/periph/spi.c

```diff
@@ -134,12 +134,20 @@ void spi_init_pins(spi_t bus)
     gpio_init_af(spi_config[bus].mosi_pin, GPIO_AF_OUT_PP);
     gpio_init(spi_config[bus].miso_pin, GPIO_IN);
 #else
-    gpio_init(spi_config[bus].mosi_pin, GPIO_OUT);
-    gpio_init(spi_config[bus].miso_pin, GPIO_IN);
-    gpio_init(spi_config[bus].sclk_pin, GPIO_OUT);
-    gpio_init_af(spi_config[bus].mosi_pin, spi_config[bus].mosi_af);
-    gpio_init_af(spi_config[bus].miso_pin, spi_config[bus].miso_af);
-    gpio_init_af(spi_config[bus].sclk_pin, spi_config[bus].sclk_af);
+    if (gpio_is_valid(spi_config[bus].mosi_pin)) {
+        gpio_init(spi_config[bus].mosi_pin, GPIO_OUT);
+        gpio_init_af(spi_config[bus].mosi_pin, spi_config[bus].mosi_af);
+    }
+
+    if (gpio_is_valid(spi_config[bus].miso_pin)) {
+        gpio_init(spi_config[bus].miso_pin, GPIO_IN);
+        gpio_init_af(spi_config[bus].miso_pin, spi_config[bus].miso_af);
+    }
+
+    if (gpio_is_valid(spi_config[bus].sclk_pin)) {
+        gpio_init(spi_config[bus].sclk_pin, GPIO_OUT);
+        gpio_init_af(spi_config[bus].sclk_pin, spi_config[bus].sclk_af);
+    }
```

Comment:
```suggestion
#ifdef SUBGHZSPI_IRQn
    if (bus == SUBGHZSPI) {
        return;
    }
#endif
    gpio_init(spi_config[bus].mosi_pin, GPIO_OUT);
    gpio_init(spi_config[bus].miso_pin, GPIO_IN);
    gpio_init(spi_config[bus].sclk_pin, GPIO_OUT);
    gpio_init_af(spi_config[bus].mosi_pin, spi_config[bus].mosi_af);
    gpio_init_af(spi_config[bus].miso_pin, spi_config[bus].miso_af);
    gpio_init_af(spi_config[bus].sclk_pin, spi_config[bus].sclk_af);
```

Reply:
The catch here is what if I am using both `PERIPH` and `GPIO` ?

Reply:
This was also to future proof if I am not wrong.

Reply:
If there are other use cases for no MOSI/MISO spi other than the subghzspi then that would make sense.

Reply:
MOSI only could be used for LEDs

MISO only could be used for Gamepads

Reply:
Ok, dismissing my comment